src/taipy/core/data/_data_repository.py

```diff
@@ -205,6 +208,10 @@ def _from_model(self, model: _DataNodeModel):
         validity_period = None
         if model.validity_seconds is not None and model.validity_days is not None:
             validity_period = timedelta(days=model.validity_days, seconds=model.validity_seconds)
+
+        if model.scope == Scope.PIPELINE:
+            _warn_deprecated(Scope.PIPELINE, Scope.SCENARIO)
```

Comment:
WHy do you use Scope enum object here, and string in data_node warning?

Reply:
Oh you're right, I noticed the deprecated value is expecting a str, so I keep it as str in data_node. But you're right, we can simply pass the scope value instead

Reply:
same for in task.py

Reply:
I have changed it to ```_warn_deprecated(f"`{Scope.PIPELINE}`", suggest="other `Scope` value")``` to be consistent with the config checker message